imcache.go

```diff
@@ -196,7 +199,7 @@ func (c *Cache[K, V]) GetMultiple(keys ...K) map[K]V {
 // the Replace method instead.
 func (c *Cache[K, V]) Set(key K, val V, exp Expiration) {
 	now := time.Now()
-	newEntry := entry[K, V]{val: val}
+	newEntry := entry[K, V]{key: key, val: val}
 	exp.apply(&newEntry.exp)
 	newEntry.SetDefaultOrNothing(now, c.defaultExp, c.sliding)
```

Comment:
Encapsulate the creation of new entry. Too many steps including setting expiration (and separate to set default?), it does not improve the readability.